internal/app/api/v1/testtriggers.go

```diff
@@ -0,0 +1,189 @@
+		if err != nil {
+			return s.Error(c, http.StatusBadGateway, err)
+		}
+
+		// map TestSuite but load spec only to not override metadata.ResourceVersion
+		testTriggerSpec := testtriggersmapper.MapTestTriggerUpsertRequestToTestTriggerCRD(request)
+		testTrigger.Spec = testTriggerSpec.Spec
+		testTrigger.Labels = request.Labels
+		updated, err := s.TestKubeClientset.TestsV1().TestTriggers(namespace).Update(c.UserContext(), testTrigger, v1.UpdateOptions{})
+
+		s.Metrics.IncUpdateTestTrigger(err)
+
+		if err != nil {
+			return s.Error(c, http.StatusBadGateway, err)
+		}
+
+		return c.JSON(updated)
+	}
+}
+
+// GetTestTriggerHandler is a handler for getting TestTrigger object
+func (s TestkubeAPI) GetTestTriggerHandler() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		namespace := c.Query("namespace", s.Namespace)
+		name := c.Params("id")
+		crdTestTrigger, err := s.TestKubeClientset.TestsV1().TestTriggers(namespace).Get(c.UserContext(), name, v1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return s.Warn(c, http.StatusNotFound, err)
+			}
+
+			return s.Error(c, http.StatusBadGateway, err)
+		}
+
+		testTrigger := testtriggersmapper.MapCRDToAPI(crdTestTrigger)
+		if c.Accepts(mediaTypeJSON, mediaTypeYAML) == mediaTypeYAML {
+			data, err := crd.GenerateYAML(crd.TemplateTestTrigger, []testkube.TestTrigger{testTrigger})
+			return s.getCRDs(c, data, err)
+		}
+
+		return c.JSON(testTrigger)
+	}
+}
+
+// DeleteTestTriggerHandler is a handler for deleting TestTrigger by id
+func (s TestkubeAPI) DeleteTestTriggerHandler() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		namespace := c.Query("namespace", s.Namespace)
+		name := c.Params("id")
+		err := s.TestKubeClientset.TestsV1().TestTriggers(namespace).Delete(c.UserContext(), name, v1.DeleteOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return s.Warn(c, http.StatusNotFound, err)
+			}
+
+			return s.Error(c, http.StatusBadGateway, err)
+		}
+
+		return c.SendStatus(http.StatusNoContent)
+	}
+}
+
+// DeleteTestTriggersHandler is a handler for deleting all or selected TestTriggers
+func (s TestkubeAPI) DeleteTestTriggersHandler() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		var err error
+		var testTriggerNames []string
+
+		namespace := c.Query("namespace", s.Namespace)
+		selector := c.Query("selector")
+		if selector == "" {
+			err = s.TestKubeClientset.TestsV1().TestTriggers(namespace).DeleteCollection(c.UserContext(), v1.DeleteOptions{}, v1.ListOptions{})
+		} else {
+			listOpts := v1.ListOptions{LabelSelector: selector}
+			testTriggerList, err := s.TestKubeClientset.TestsV1().TestTriggers(namespace).List(c.UserContext(), listOpts)
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					return s.Error(c, http.StatusBadGateway, err)
+				}
+			} else {
+				for _, item := range testTriggerList.Items {
+					testTriggerNames = append(testTriggerNames, item.Name)
```

Comment:
How do you use testTriggerNames?

Reply:
actually this was done in a hurry, I refactored it